packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -930,17 +956,15 @@ export class BasicCrawler<
             throw error;
         }
 
+        // @ts-expect-error It is assignable, but TS says otherwise...
+        const castedErrorContext = crawlingContext as ErrorContext;
+        castedErrorContext.error = error;
+
         const shouldRetryRequest = !request.noRetry && request.retryCount < this.maxRequestRetries && !(error instanceof NonRetryableError);
         if (shouldRetryRequest) {
             request.retryCount++;
-            const { url, retryCount, id } = request;
-            // We don't want to see the stack trace in the logs by default, when we are going to retry the request.
-            // Thus, we print the full stack trace only when CRAWLEE_VERBOSE_LOG environment variable is set to true.
-            this.log.warning(
-                `Reclaiming failed request back to the list or queue. ${!process.env.CRAWLEE_VERBOSE_LOG ? error : error.stack}`,
-                { id, url, retryCount },
-            );
-            await source.reclaimRequest(request);
+
+            await this._handlePreReclaimRequestHandler(castedErrorContext, source);
```

Comment:
the code here should stay as it was, just execute the fail handler before it (but after incrementing the retry count)

Reply:
So basically we would add another option to crawler, but it would not be top-level function, but instead would be called instead of the currently existing one, correct?

Reply:
or actually let me re-read it :)